File: backend/src/main/java/com/antshorttv/ai/EditableAiDefinitionController.java
package com.antshorttv.ai;

import com.antshorttv.common.ApiResponse;
import com.antshorttv.platform.RequirePlatformPermission;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.validation.Valid;
import java.util.List;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/platform/ai/definitions")
public class EditableAiDefinitionController {
    private final EditableAiDefinitionService service;
    public EditableAiDefinitionController(EditableAiDefinitionService service) { this.service = service; }
    @GetMapping("/agents") @RequirePlatformPermission("PLATFORM_AI_AGENT_VIEW")
    public ApiResponse<List<EditableAgentResponse>> agents() { return ApiResponse.success(service.agents()); }
    @PutMapping("/agents/{code}") @RequirePlatformPermission("PLATFORM_AI_AGENT_EDIT")
    public ApiResponse<EditableAgentResponse> updateAgent(@PathVariable String code, @Valid @RequestBody EditableAgentRequest body, HttpServletRequest request) { return ApiResponse.success(service.updateAgent(code, body, request)); }
    @PostMapping("/agents/{code}/preview") @RequirePlatformPermission("PLATFORM_AI_AGENT_VIEW")
    public ApiResponse<String> previewAgent(@PathVariable String code, @RequestBody(required = false) java.util.Map<String, Object> variables) { return ApiResponse.success(service.previewAgent(code, variables)); }
    @PostMapping("/agents/{code}/publish") @RequirePlatformPermission("PLATFORM_AI_AGENT_EDIT")
    public ApiResponse<EditableAgentResponse> publishAgent(@PathVariable String code, HttpServletRequest request) { return ApiResponse.success(service.publishAgent(code, request)); }
    @PostMapping("/agents/{code}/{status}") @RequirePlatformPermission("PLATFORM_AI_AGENT_EDIT")
    public ApiResponse<EditableAgentResponse> setAgentStatus(@PathVariable String code, @PathVariable String status, HttpServletRequest request) { return ApiResponse.success(service.setAgentStatus(code, status, request)); }
    @PostMapping("/agents/{code}/rollback/{version}") @RequirePlatformPermission("PLATFORM_AI_AGENT_EDIT")
    public ApiResponse<EditableAgentResponse> rollbackAgent(@PathVariable String code, @PathVariable Integer version, HttpServletRequest request) { return ApiResponse.success(service.rollbackAgent(code, version, request)); }
    @GetMapping("/skills") @RequirePlatformPermission("PLATFORM_AI_AGENT_VIEW")
    public ApiResponse<List<EditableSkillResponse>> skills() { return ApiResponse.success(service.skills()); }
    @PutMapping("/skills/{code}") @RequirePlatformPermission("PLATFORM_AI_SKILL_EDIT")
    public ApiResponse<EditableSkillResponse> updateSkill(@PathVariable String code, @Valid @RequestBody EditableSkillRequest body, HttpServletRequest request) { return ApiResponse.success(service.updateSkill(code, body, request)); }
    @PostMapping("/skills/{code}/publish") @RequirePlatformPermission("PLATFORM_AI_SKILL_EDIT")
    public ApiResponse<EditableSkillResponse> publishSkill(@PathVariable String code, HttpServletRequest request) { return ApiResponse.success(service.publishSkill(code, request)); }
    @PostMapping("/skills/{code}/{status}") @RequirePlatformPermission("PLATFORM_AI_SKILL_EDIT")
    public ApiResponse<EditableSkillResponse> setSkillStatus(@PathVariable String code, @PathVariable String status, HttpServletRequest request) { return ApiResponse.success(service.setSkillStatus(code, status, request)); }
    @PostMapping("/skills/{code}/rollback/{version}") @RequirePlatformPermission("PLATFORM_AI_SKILL_EDIT")
    public ApiResponse<EditableSkillResponse> rollbackSkill(@PathVariable String code, @PathVariable Integer version, HttpServletRequest request) { return ApiResponse.success(service.rollbackSkill(code, version, request)); }
}

File: frontend/src/services/ant-design-pro/editableAiDefinitionController.ts
// @ts-ignore
/* eslint-disable */
import { request } from "@umijs/max";

/** 此处后端没有提供注释 GET /api/platform/ai/definitions/agents */
export async function agents(options?: { [key: string]: any }) {
  return request<API.ApiResponseListEditableAgentResponse>(
    "/api/platform/ai/definitions/agents",
    {
      method: "GET",
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 PUT /api/platform/ai/definitions/agents/${param0} */
export async function updateAgent(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.updateAgentParams,
  body: API.EditableAgentRequest,
  options?: { [key: string]: any }
) {
  const { code: param0, ...queryParams } = params;
  return request<API.ApiResponseEditableAgentResponse>(
    `/api/platform/ai/definitions/agents/${param0}`,
    {
      method: "PUT",
      headers: {
        "Content-Type": "application/json",
      },
      params: { ...queryParams },
      data: body,
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 POST /api/platform/ai/definitions/agents/${param0}/${param1} */
export async function setAgentStatus(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.setAgentStatusParams,
  options?: { [key: string]: any }
) {
  const { code: param0, status: param1, ...queryParams } = params;
  return request<API.ApiResponseEditableAgentResponse>(
    `/api/platform/ai/definitions/agents/${param0}/${param1}`,
    {
      method: "POST",
      params: { ...queryParams },
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 POST /api/platform/ai/definitions/agents/${param0}/preview */
export async function previewAgent(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.previewAgentParams,
  options?: { [key: string]: any }
) {
  const { code: param0, ...queryParams } = params;
  return request<API.ApiResponseString>(
    `/api/platform/ai/definitions/agents/${param0}/preview`,
    {
      method: "POST",
      params: { ...queryParams },
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 POST /api/platform/ai/definitions/agents/${param0}/publish */
export async function publishAgent(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.publishAgentParams,
  options?: { [key: string]: any }
) {
  const { code: param0, ...queryParams } = params;
  return request<API.ApiResponseEditableAgentResponse>(
    `/api/platform/ai/definitions/agents/${param0}/publish`,
    {
      method: "POST",
      params: { ...queryParams },
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 POST /api/platform/ai/definitions/agents/${param0}/rollback/${param1} */
export async function rollbackAgent(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.rollbackAgentParams,
  options?: { [key: string]: any }
) {
  const { code: param0, version: param1, ...queryParams } = params;
  return request<API.ApiResponseEditableAgentResponse>(
    `/api/platform/ai/definitions/agents/${param0}/rollback/${param1}`,
    {
      method: "POST",
      params: { ...queryParams },
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 GET /api/platform/ai/definitions/skills */
export async function skills1(options?: { [key: string]: any }) {
  return request<API.ApiResponseListEditableSkillResponse>(
    "/api/platform/ai/definitions/skills",
    {
      method: "GET",
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 PUT /api/platform/ai/definitions/skills/${param0} */
export async function updateSkill(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.updateSkillParams,
  body: API.EditableSkillRequest,
  options?: { [key: string]: any }
) {
  const { code: param0, ...queryParams } = params;
  return request<API.ApiResponseEditableSkillResponse>(
    `/api/platform/ai/definitions/skills/${param0}`,
    {
      method: "PUT",
      headers: {
        "Content-Type": "application/json",
      },
      params: { ...queryParams },
      data: body,
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 POST /api/platform/ai/definitions/skills/${param0}/${param1} */
export async function setSkillStatus(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.setSkillStatusParams,
  options?: { [key: string]: any }
) {
  const { code: param0, status: param1, ...queryParams } = params;
  return request<API.ApiResponseEditableSkillResponse>(
    `/api/platform/ai/definitions/skills/${param0}/${param1}`,
    {
      method: "POST",
      params: { ...queryParams },
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 POST /api/platform/ai/definitions/skills/${param0}/publish */
export async function publishSkill(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.publishSkillParams,
  options?: { [key: string]: any }
) {
  const { code: param0, ...queryParams } = params;
  return request<API.ApiResponseEditableSkillResponse>(
    `/api/platform/ai/definitions/skills/${param0}/publish`,
    {
      method: "POST",
      params: { ...queryParams },
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 POST /api/platform/ai/definitions/skills/${param0}/rollback/${param1} */
export async function rollbackSkill(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.rollbackSkillParams,
  options?: { [key: string]: any }
) {
  const { code: param0, version: param1, ...queryParams } = params;
  return request<API.ApiResponseEditableSkillResponse>(
    `/api/platform/ai/definitions/skills/${param0}/rollback/${param1}`,
    {
      method: "POST",
      params: { ...queryParams },
      ...(options || {}),
    }
  );
}
